### PR TITLE
ib, sdk: use zstd for compression

### PR DESCRIPTION
**Pull request by aparcar**
Use similar efficient but faster compression algorithm.

### CODE REVIEW

**Review by systemcrash — APPROVED** [review]

**Comment by aparcar** [comment]
@neocturne didnt you do some benchmarks in the past?

**Comment by neocturne** [comment]
> @neocturne didnt you do some benchmarks in the past?

I did some tests, but I haven't written down any concrete numbers. See https://patchwork.ozlabs.org/project/openwrt/patch/0f48abf2ac872957d6a4a150ead39564053f2afc.1589716209.git.mschiffer@universe-factory.net/ for previous discussion.

**Comment by systemcrash** [comment]
I've run numbers a few times. Zstd doesn't always win in compressed size,
but where it shines is extraction speed: it's orders of magnitude faster.
Same thing for "good enough" compression.

If the goal is size on on openwrt installation, xz still wins iirc.

**Comment by Ansuel** [comment]
> I've run numbers a few times. Zstd doesn't always win in compressed size, but where it shines is extraction speed: it's orders of magnitude faster. Same thing for "good enough" compression. If the goal is size on an openwrt installation, xz still wins iirc.

well sdk and ib we don't care about size but actually extraction speed... so it might be good to use xz...

File: target/imagebuilder/Makefile [hunk 1]
```diff
@@ -24,7 +24,7 @@ BUNDLER_COMMAND := PATH=$(BUNDLER_PATH) $(XARGS) $(SCRIPT_DIR)/bundle-libraries.
 
 all: compile
 
-$(BIN_DIR)/$(IB_NAME).tar.xz: clean
+$(BIN_DIR)/$(IB_NAME).tar.zstd: clean
```
Comment (aparcar) [inline target/imagebuilder/Makefile:27]: should end on `.zst`

**Comment by robimarko** [comment]
As @f00b4r0 pointed out here are some measurements that Arch did few years back: https://lists.archlinux.org/pipermail/arch-dev-public/2019-March/029520.html

**Comment by systemcrash** [comment]
> As @f00b4r0 pointed out here are some measurements that Arch did few years back: https://lists.archlinux.org/pipermail/arch-dev-public/2019-March/029520.html

Damn swedes and their commas in the wrong places 😆

**Review by f00b4r0 — APPROVED** [review]
tested working on top of 076e860. Let's ship it :)
NB: llvm and toolchain are still packaged as xz

**Comment by f00b4r0** [comment]
One important bit: pushing this PR must be sync'd with updating https://github.com/openwrt/buildbot/blob/65502c7ec51968435036ab43be208fcb08728a5c/phase2/master.cfg#L118 (on whatever codebase phase2 is running)

**Comment by robimarko** [comment]
@f00b4r0 switched toolchain to ZSTD as well

**Comment by Ansuel** [comment]
btw i think also wiki needs to be updated?

**Comment by f00b4r0** [comment]
@robimarko thanks, only target/llvm-bpf/Makefile is left I think
@Ansuel what do you have in mind?

**Comment by Ansuel** [comment]
About this or llvm?

**Comment by robimarko** [comment]
LLVM BPF is now ZSTD compressed as well though I cannot test this because LLVM wont compile for me

**Comment by f00b4r0** [comment]
> LLVM BPF is now ZSTD compressed as well though I cannot test this because LLVM wont compile for me

Tested working on top of 7f85104e

**Comment by f00b4r0** [comment]
> About this or llvm?

Got it, at least these pages will need to be updated:
- https://openwrt.org/docs/guide-user/additional-software/imagebuilder
- https://openwrt.org/docs/guide-developer/toolchain/using_the_sdk

There may be more, maybe it's possible for an admin grep the wiki to look for '.tar.xz'?

We should probably put out a statement explaining that we want to move away from xz, so that package maintainers may prefer other archive formats when available. This page would probably need to reflect that: https://openwrt.org/docs/guide-developer/packages

**Comment by systemcrash** [comment]
Future: can we have docs.openwrt.org? All documentation stubs could go
there.

**Comment by ynezz** [comment]
> One important bit: pushing this PR must be sync'd with updating https://github.com/openwrt/buildbot/blob/65502c7ec51968435036ab43be208fcb08728a5c/phase2/master.cfg#L118 (on whatever codebase phase2 is running)

Done.

**Pull request merged**